### PR TITLE
🧹 Improve MyST CLI interfaces for export

### PR DESCRIPTION
**Pull request by fwkoch**
This is not a pretty PR with a large diff, but it's scope is relatively small and works to make things nicer for developers in a few ways:

1. `Options` types are cleaned up so they have useful names and are appropriately reused - no more amalgamated `Options` types that serve 3 purposes.
2. Export functions that are not used anywhere in the myst ecosystem  (but are still used externally) are moved to a `legacy` file
3. `myst-cli` exports some reusable cli interfaces, so parallel tools that consume myst can stay in-sync with available options.

### CODE REVIEW

File: packages/myst-cli/src/index.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 export * from './build/index.js';
+export * from './cli/index.js';
```
Comment (rowanc1) [inline packages/myst-cli/src/index.ts:2]: Nice -- good to see these being made more reusable. 🚀

File: packages/myst-cli/src/build/legacy.ts [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+  watch?: boolean;
+  throwOnFailure?: boolean;
+  ci?: boolean;
+  renderer?: RendererFn;
+};
+
+// These "local article" export functions are not used across the MyST ecosystem.
+// However, they are still maintained as exports for external libraries.
+
+export async function localArticleToWord(
```
Comment (fwkoch) [inline packages/myst-cli/src/build/legacy.ts:42]: Add deprecated annotation to these as well.

**Review by rowanc1 — APPROVED** [review]
Spent a few hours going through this with @fwkoch in person. And have tested init/build/start in myst and no problems found.

I think the only thing to add is the `@deprecated` label on the legacy functions.

File: packages/myst-cli/src/index.ts (hunk 1 above)
Comment (rowanc1) [reply packages/myst-cli/src/index.ts:2]: It is really good to see the options actually correct as well.

**Pull request merged**